sprite.h

```diff
@@ -29,20 +29,41 @@ class Sprite : public Identifiable<Empty> {
 	}
 
 	Point &home();
+	inline static bool allow_screen_wrapping = true;
 
 protected:
 	void transform();
 	void update_trail();
 	void increment_trail_index(const size_t amount = 1);
 	TrailSprite& get_trail(const size_t index = 0);
 	virtual void draw_trail(Context &ctx);
+	void randomize_tendency();
 
 	const Texture *m_texture;
 	Point m_relpos;
 	Point m_home;
 	GLdouble m_size;
 	std::vector<TrailSprite> m_trail;
 	size_t m_trail_start_index;
+
+	template<typename T>							// first value gets multiplied or divided by 1 + random(second value)
+	using pair_randomness = std::pair<T, double>;	// probably needs a better name :)
+
+	template<typename T>
+	double randomize_tendency_variable(pair_randomness<T> tendency_pair);
+
+	inline const static pair_randomness<size_t> default_frames_between_tendency_changes = { 10 * 30, 0.5 };
+	inline const static pair_randomness<double> default_tendency_distance_from_home_bias = { 1.5, 0.2 };
+	inline const static pair_randomness<double> default_wiggle_amount_bias = { 1.0, 0.25 };
+	inline const static pair_randomness<double> default_distance_from_tendency_bias = { 2.5, 1.0 };
+	inline const static double shake_divisor = 1 / 0.3;
+	inline const static double home_drift_divisor = 10.0;
```

Comment:
I don't have any problem with doing some complicated math to make more interesting randomness, but this complexity is bleeding all over the place. This sprite class now looks waaaay more complicated than it really is because there are a dozen or more members flying all over the fucking place (just look at the constructor of this class, my god.) If we're going to make wiggling and tendency so complicated, then let's stuff all that complexity into a box that we can ignore, because otherwise in the future we're going to come back to this class and it's going to totally overwhelm us. A sprite is really simple, we shouldn't be overwhelmed.

I think the best approach here is to make a SpriteWiggler class that becomes a single member of Sprite which is itself responsible for all this crap:
* all five of the instance variables you added just below here as well as 
* all these statics above, plus the templates
* the `randomize_tendency` and related functions
 
Then, in our `Sprite::update` function, we can just call `m_home = m_sprite_wiggler.wiggle(m_home)` or something like that and be fucking done with it.